frontend/src/views/components/LayersList/Layer/Layer.component.jsx

```diff
@@ -72,7 +99,12 @@ const Layer = ({
           className="panel-input-switch"
           id={`layer_${id}`}
           checked={isActive}
-          onChange={toggleLayer}
+          onChange={() => {
+            toggleLayer();
+            if (!isActive) {
+              sendAnalyticsEvent();
+            }
```

Comment:
I think the layer is not yet active so `isActive` is false.

Reply:
ok, I see...  So, in that case, it should be inside the `useCallback` function called `toggleLayer` in L46. I think it will be more consistent and you can check `isActive` inside

Reply:
But then the function will change every time we toggle the layer on that case so useCallback won't have sense anymore. Anyway, we want to check if the layer is not active when we hit the toggle button so it always will be !isActive

Reply:
```
           onChange={() => {
            toggleLayer();
            if (!isActive) {
              sendAnalyticsEvent();
            }
```
Yeah, but in this code, it will change in every render. Is ok, next time I'll make a better suggestion code to explain better my point